### PR TITLE
feat(http-core): add raw description to error response

### PR DESCRIPTION
**Pull request by aimeex3**

### CODE REVIEW

File: packages/http-core/src/http-error.js [hunk 1]
```diff
@@ -91,7 +102,8 @@ const HttpError = extendError({
     return message;
   },
   properties: {
-    defaultMessage: `An error was received while trying to fulfill the request`
+    defaultMessage: `An error was received while trying to fulfill the request`,
+    defaultRawDescription: `none`
```
Comment (ianwremmel) [inline packages/http-core/src/http-error.js]: why `none` rather than `\`\``?

File: packages/http-core/src/http-error.js [hunk 2]
```diff
@@ -122,6 +134,36 @@ function parseObject(body) {
   return message;
 }
 
+/**
+ * @param {object} body
+ * @private
+ * @returns {string}
+ */
+function parseDescription(body) {
```
Comment (ianwremmel) [inline packages/http-core/src/http-error.js]: I think only SparkHttpError should have the rawDescription field, since it's the one that adds the tracking id and url to the message. on that note, `rawMessage` might be a better field name.

File: packages/spark-core/src/lib/spark-http-error.js [hunk 3]
```diff
@@ -6,17 +6,43 @@
...
 import {HttpError} from '@ciscospark/http-core';
+import {isArray, pick} from 'lodash';
 
 const SparkHttpError = extendError(HttpError, {
   parseFn(res, ...rest) {
+    /* eslint complexity: [0] */
     /* eslint prefer-reflect: [0] */
     let message = HttpError.prototype.parseFn.call(this, res, ...rest);
 
+    let body = res.body;
```
Comment (ianwremmel) [inline packages/spark-core/src/lib/spark-http-error.js]: Why are these changes necessary? Doesn't HttpError already do this?

Comment (aimeex3) [inline packages/spark-core/src/lib/spark-http-error.js]: This parses just the description out of the error. HttpError parses the entire message. Initially I added the description parsing as part of HttpError, but you mentioned that it should be moved to SparkHttpError instead

**Comment by ianwremmel** [comment]
can this be closed?

**Comment by aimeex3** [comment]
yeah this one may need to get revisited for the new sdk
